ls1intum/Artemis
# [Feature] Enable static code analysis for GitLab + Jenkins

### Checklist

### Motivation and Context


This PR enables the handling of static code analysis on Artemis side. The Jenkins plugin was already adapted to support the parsing and sending of issues found by static code analysis.

### Description


- If static code analysis was enabled, a build plan with an additional post build task will be created. The task runs static code analysis and copies the reports to a specific folder. The plugin will look for reports in this folder.
- Extend `TestResultDTO` to include static code analysis reports sent by the plugin
- Extend `JenkinsService` to create static code analysis feedback if the option was enabled and such feedback was sent by the plugin

### Steps for Testing

**Use TS2 for testing. The new version of the notification plugin was already deployed there for testing purposes**

1. Log in to Artemis Test Server 2
2. Create a programming exercise with and without static code analysis
3. Participate in the exercises
4. Submit a solution
-> Check the you receive the expected feedback. For the exercise with static code analysis enabled, such feedback should be visible

### Test Coverage

* ExerciseService.java: 85%
* programming-exercise.component.ts 95%
